tests/unit/test-class-wc-subscriptions.php

```diff
@@ -1137,13 +1137,13 @@ public function test_update_status_to_pending_canellation() {
 			if ( in_array( $status, $expected_to_pass, true ) ) {
 
 				try {
-
-					$start_date = gmdate( 'Y-m-d H:i:s', strtotime( '-1 month' ) );
+					$expected_end_date = gmdate( 'Y-m-d H:i:s' );
+					$start_date        = gmdate( 'Y-m-d H:i:s', strtotime( $expected_end_date . ' -1 month' ) );
```

Comment:
I've tested this by adding a sleep(1); between lines 1141-1142 and 1142-1143 and in both cases they fail

Reply:
For reference, here's the updated test I had:

```
$start_time        = time();
$expected_end_time = strtotime( '+1 month', $start_time );

$subscription->update_dates( [
	'start'        => gmdate( 'Y-m-d H:i:s', $start_time ),
	'next_payment' => gmdate( 'Y-m-d H:i:s', $expected_end_time ),
] );

$subscription->update_status( 'pending-cancel' );

$this->assertEquals( $expected_end_time, $subscription->get_time( 'end' ) );
```

Reply:
Fixed in [a218e45](https://github.com/Automattic/woocommerce-subscriptions-core/pull/159/commits/a218e4549b1017bce6c82aecfccdc78d03ace860)